### PR TITLE
[#38] Implement exception handling

### PR DESCRIPTION
**Pull request by Sheptytskyid**

### CODE REVIEW

File: src/main/java/com/manardenza/dao/AbstractDao.java [hunk 1]
```diff
@@ -27,17 +29,19 @@ private void saveToFile() {
         try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(databaseFile))) {
             oos.writeObject(database);
         } catch (IOException e) {
-            log.error("Error while attempting to write to database file", e);
-            //FIXME: rethrow to console.
+            String message = "Error while attempting to read from database file";
+            log.error(message, e);
```
Comment (zozich) [inline src/main/java/com/manardenza/dao/AbstractDao.java]: It is better to log original exception message

File: src/main/java/com/manardenza/dao/AbstractDao.java [hunk 2]
```diff
@@ -27,17 +29,19 @@ private void saveToFile() {
...
     }
 
     private void readFromFile() {
         try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(databaseFile))) {
             database = (List<T>) ois.readObject();
         } catch (IOException | ClassNotFoundException e) {
-            log.error("Error while attempting to read from database file", e);
-            //FIXME: rethrow to console.
+            String message = "Error while attempting to read from database file";
+            log.error(message, e);
```
Comment (zozich) [inline src/main/java/com/manardenza/dao/AbstractDao.java]: Same

File: src/main/java/com/manardenza/exceptions/DbFileReadException.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.manardenza.exceptions;
+
+public class DbFileReadException extends RuntimeException {
+
+    public DbFileReadException(String message, Throwable cause) {
```
Comment (zozich) [inline src/main/java/com/manardenza/exceptions/DbFileReadException.java]: 1. Why do you need two separate exceptions for similar operation?
2. Why do you need cause here?

**Review by zozich — CHANGES_REQUESTED** [review]

File: src/main/java/com/manardenza/exceptions/DbFileReadException.java (hunk 3 above)
Comment (Sheptytskyid) [inline src/main/java/com/manardenza/exceptions/DbFileReadException.java]: to print into console

Comment (Sheptytskyid) [inline src/main/java/com/manardenza/exceptions/DbFileReadException.java]: 1. agree removed
2. to get the full stack trace

**Review by zozich — APPROVED** [review]
Ok

**Pull request merged**